core/trino-main/src/main/java/io/trino/operator/DynamicFilterSourceOperator.java

```diff
@@ -143,7 +143,32 @@ public void noMoreOperators()
         @Override
         public OperatorFactory duplicate()
         {
-            throw new UnsupportedOperationException("duplicate() is not supported for DynamicFilterSourceOperatorFactory");
+            // A duplicate factory may be required for DynamicFilterSourceOperatorFactory in fault tolerant execution
+            // by LocalExecutionPlanner#addLookupOuterDrivers to add a new driver to output the unmatched rows in an outer join
+            return new DynamicFilterSourceOperatorFactory(
+                    operatorId,
+                    planNodeId,
+                    new DynamicFilterSourceConsumer() {
```

Comment:
@arhimondr @sopel39 I'm assuming here that the operator from duplicate factory doesn't need to collect dynamic filters.
Let me know if that is valid assumption.

Reply:
Why it doesn't need to collect anything (from outer right join)? I'm missing context here.

Outer unmatched rows are different than matched left rows

Reply:
This is related to the code in `LocalExecutionPlanner#addLookupOuterDrivers`, I don't fully understand what that is about. It says "Add a new driver to output the unmatched rows in an outer join.". So it sounded like the DynamicFilterSourceOperator that comes from the duplicate factory doesn't need to care about these "unmatched rows" (does it ?).

Reply:
Check how right outer join is planned.
There are actually two downstream pipelines (after outer join). One is the main one for matched rows. The other is duplicated for unmatched outer (right) rows. I assume that since dynamic filter operator is downstream then these unmatched rows are important.

Have you added test where DF contains unmatched outer rows. That's where I expect incorrect results.

Reply:
Okay, I think I get the problem. So we can't ignore unmatched rows.
However, duplicating the factory breaks the existing logic around tracking expected partitions count for the DF consumer through `noMoreOperators` (consumer would receive `setPartitionCount` from 2 different factories). So I think I'm just going to turn off DF for this specific scenario to avoid query failure and any correctness issue.

Reply:
I pushed changes to turn off DF in this scenario instead of now, PTAL

Reply:
Filed https://github.com/trinodb/trino/issues/15628